charts/karpenter/templates/post-install-hook.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     {{- include "karpenter.labels" . | nindent 4 }}
   annotations:
-    "helm.sh/hook": post-install
+    "helm.sh/hook": post-install,post-upgrade,post-rollback
```

Comment:
This feels dangerous to me. Users installing this chart won't actually get management of their CRDs when they do the upgrade. Helm doesn't upgrade CRDs going from version to version, so I think you only need this on the installation path

Reply:
Discussed offline: After playing with it for a bit, I realized that a user that is going to be executing this hook is going to be going through the process of kubectl applying anyways. By adding the hook in here on rollback and upgrade, this is going to make it so that a user doesn't even have to think about the webhook port or namespace when applying the karpenter helm chart.